powerscale/helper/nfs_zone_settings_helper.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+	powerscale "dell/powerscale-go-client"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/models"
+)
+
+// FilterNfsZoneSettings filter nfs zone settings.
+func FilterNfsZoneSettings(ctx context.Context, client *client.Client, filter *models.NfsZoneSettingsFilter) (*powerscale.V2NfsSettingsZone, error) {
+	filterParam := client.PscaleOpenAPIClient.ProtocolsApi.GetProtocolsv2NfsSettingsZone(ctx)
+
+	if filter != nil {
```

Comment:
how about` if filter != nil && filter.Zone.ValueString()  != "" `directly

Reply:
Yes, that can also be done in this case where there is only one query parameter i.e. zone.
But I kept it that way to keep it more general where if there are multiple query parameters e.g. zone, scope, then in such a case we can first have a check for the filter and then within that checks for each query parameter.

Reply:
thanks @MangirishK , understood